sanic/server/websockets/impl.py

```diff
@@ -0,0 +1,776 @@
+        like :meth:`recv`.
+        """
+        if self.recv_lock.locked():
+            raise ServerError(
+                "Cannot call recv_streaming while another task "
+                "is already waiting for the next message"
+            )
+        await self.recv_lock.acquire()
+        if self.connection.state in (CLOSED, CLOSING):
+            raise ServerError(
+                "Cannot receive from websocket interface after it is closed."
+            )
+        try:
+            self.can_pause = False
+            async for m in self.assembler.get_iter():
+                yield m
+        finally:
+            self.can_pause = True
+            self.recv_lock.release()
+
+    async def send(self, message: Union[Data, Iterable[Data]]) -> None:
+        """
+        Send a message.
+        A string (:class:`str`) is sent as a `Text frame`_. A bytestring or
+        bytes-like object (:class:`bytes`, :class:`bytearray`, or
+        :class:`memoryview`) is sent as a `Binary frame`_.
+        .. _Text frame: https://tools.ietf.org/html/rfc6455#section-5.6
+        .. _Binary frame: https://tools.ietf.org/html/rfc6455#section-5.6
+        :meth:`send` also accepts an iterable of strings, bytestrings, or
+        bytes-like objects. In that case the message is fragmented. Each item
+        is treated as a message fragment and sent in its own frame. All items
+        must be of the same type, or else :meth:`send` will raise a
+        :exc:`TypeError` and the connection will be closed.
+        :meth:`send` rejects dict-like objects because this is often an error.
+        If you wish to send the keys of a dict-like object as fragments, call
+        its :meth:`~dict.keys` method and pass the result to :meth:`send`.
+        :raises TypeError: for unsupported inputs
+        """
+        async with self.conn_mutex:
+
+            if self.connection.state in (CLOSED, CLOSING):
+                raise ServerError(
+                    "Cannot write to websocket interface after it is closed."
+                )
+            if self.data_finished_fut is None or self.data_finished_fut.done():
+                raise ServerError(
+                    "Cannot write to websocket interface after it is finished."
+                )
+
+            # Unfragmented message -- this case must be handled first because
+            # strings and bytes-like objects are iterable.
+
+            if isinstance(message, str):
+                self.connection.send_text(message.encode("utf-8"))
+                await self.send_data(self.connection.data_to_send())
+
+            elif isinstance(message, (bytes, bytearray, memoryview)):
+                self.connection.send_binary(message)
+                await self.send_data(self.connection.data_to_send())
+
+            elif isinstance(message, Mapping):
+                # Catch a common mistake -- passing a dict to send().
+                raise TypeError("data is a dict-like object")
```

Comment:
:tada:

Reply:
This was implemented as suggested by @aaugustin